### PR TITLE
docs/checkbox hooks page

### LINKED ISSUE #72: (docs): create a hooks page for the Checkbox hooks
In the Checkbox documentation there is a section about hooks. This one only mentions `useCheckbox` and `useCheckboxGroup` but does not give any information on how they should be used or an example.

Here is a link to the section: https://chakra-ui.com/docs/form/checkbox#hooks

They should include the return values, parameters, and at least one usage case as an example.

> Moderator note: we should create a new page for the checkbox hooks in the "Hooks" section, but reference it in the Checkbox component page.

### PR DESCRIPTION
**Pull request by noobinthisgame**
Closes #72 

## 📝 Description

Added two new pages for the useCheckbox and useCheckboxGroup hooks that describe their functionality. Linked the pages in the checkbox documentation.

## ⛳️ Current behavior (updates)

Only a small description about the hooks in the Checkbox documentation.

## 🚀 New behavior

Two new detailed pages that explain how the useCheckbox and useCheckboxGroup hooks can be used.

## 💣 Is this a breaking change (Yes/No):

No

### CODE REVIEW

**Comment by lazarnikolov94** [comment]
This is great @noobinthisgame! Could you also write up some examples? Here's a little codesandbox I created that can help you get started: https://codesandbox.io/s/custom-checkbox-using-usecheckbox-hook-82pgi?file=/src/index.tsx

**Comment by noobinthisgame** [comment]
Ye of course! This is still work in progress, I didn't find how zo make a draft PR. Thanks for the codesanbox! 😁

**Comment by noobinthisgame** [comment]
FYI back from vacation and back on this!

**Comment by noobinthisgame** [comment]
To make it easier to document and easier to understand I made some changes to the main code. Thus this PR is depending on those changes to be added and merged https://github.com/chakra-ui/chakra-ui/pull/5375

**Comment by noobinthisgame** [comment]
@nikolovlazar I finally found some time to get this done! :) Ready for review now. Could you please also check what happens when you change the strings to numbers in the values in the example for useCheckboxGroup. Seemed like there is something wrong when using numbers, not sure tho if it's a bug.

**Review by nikolovlazar — CHANGES_REQUESTED** [review]
@noobinthisgame this is a bug indeed. I've opened a PR in the main repo: https://github.com/chakra-ui/chakra-ui/pull/5502 that potentially fixes this issue. Thanks for letting me know!

Other than that, let's make the types of the `useCheckboxGroup` hook a bit more abstract, because they're pushing on the description, and the props table looks funky:
CleanShot 2022-02-01 at 08 39 50@2x

**Comment by noobinthisgame** [comment]
@nikolovlazar indeed those types were a bit long. I pushed some simplifying changes :)

File: pages/docs/hooks/use-checkbox.mdx [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+title: useCheckbox
+package: '@chakra-ui/checkbox'
+description: 'React hook to manage checkboxes.'
+---
+
+`useCheckbox` is a custom hook used to provide checkbox functionality, aswell as
```
Comment (nikolovlazar) [inline pages/docs/hooks/use-checkbox.mdx:7]:
```suggestion
`useCheckbox` is a custom hook used to provide checkbox functionality, as well as
```

Missing space between `as` and `well`: `aswell` -> `as well`

**Review by nikolovlazar — CHANGES_REQUESTED** [review]
Other than this typo, everything looks good and it's ready to be merged! 👌

**Comment by noobinthisgame** [comment]
Fixed

File: pages/docs/hooks/use-checkbox-group.mdx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        bg='green.50'
+        border='1px solid'
+        borderColor='green.500'
+        rounded='lg'
+        px={3}
+        py={1}
+        cursor='pointer'
+        {...htmlProps}
+      >
+        <input {...getInputProps()} hidden />
```
[the remark this answers is not in the document]
Comment (nikolovlazar) [reply pages/docs/hooks/use-checkbox-group.mdx:52]: That's interesting @julia-allyce! I tried the example csb above and I didn't get the focus as well. Can you produce a CSB example (maybe use the one above) with the focus implemented?

**Pull request merged**